### PR TITLE
[FrameworkBundle] Add support to environment variables APP_ENV/DEBUG in KernelTestCase

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/recipes/pull/170#discussion_r138425265
| License       | MIT
| Doc PR        | -

/cc @fabpot

### CODE REVIEW

**Comment by yceruto** [comment]
I guess fabbot failure (unrelated) should be fixed from `2.7`.

File: src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php [hunk 1]
```diff
@@ -188,10 +188,23 @@ protected static function createKernel(array $options = array())
             static::$class = static::getKernelClass();
         }
 
-        return new static::$class(
-            isset($options['environment']) ? $options['environment'] : 'test',
-            isset($options['debug']) ? $options['debug'] : true
-        );
+        if (isset($options['environment'])) {
+            $env = $options['environment'];
+        } elseif (isset($_SERVER['APP_ENV']) || isset($_ENV['APP_ENV'])) {
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php]: I think I would split this in two:

```php
if (isset($options['environment'])) {
    $env = $options['environment'];
} elseif (isset($_SERVER['APP_ENV'])) {
    $env = $_SERVER['APP_ENV'];
} elseif (isset($_ENV['APP_ENV'])) {
    $env = $_ENV['APP_ENV'];
} else {
    $env = 'test';
}
```

**Comment by xabbuh** [comment]
Should we merge this one as a bugfix into the `3.3` branch instead?

File: src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php]: Fixed.

**Comment by yceruto** [comment]
As a bugfix should be into `2.7`, no? Let's me know the final decision. I'm for a bugfix rather than a feature ( similar to https://github.com/symfony/symfony/pull/24115)

**Comment by fabpot** [comment]
As env vars are "officially" supported since 3.3, I'd vote for a merge in 3.3

**Review by fabpot — APPROVED** [review]

**Comment by yceruto** [comment]
Done.

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request merged**